src/utils/hooks/useTitle.ts

```diff
@@ -0,0 +1,211 @@
+// When called without an argument (from `Routes.tsx`) this hook
+// parses the window location URL to set the title. If called
+// with an argument `title` (from any component), it sets the
+// document title to `title` - this might be especially useful
+// when you want to manipulate the title even when the route
+// remains same (for example, when multiple dialogue boxes can
+// be present on the same route.)
+
+import { useLocation, matchPath } from 'react-router-dom';
+import { useEffect } from 'react';
+import { defineMessages, useIntl, MessageDescriptor } from 'react-intl';
+import {
+  CREATE_COLONY_ROUTE,
+  CONNECT_ROUTE,
+  CREATE_USER_ROUTE,
+  INBOX_ROUTE,
+  USER_EDIT_ROUTE,
+  WALLET_ROUTE,
+  NOT_FOUND_ROUTE,
+  LANDING_PAGE_ROUTE,
+  COLONY_HOME_ROUTE,
+  COLONY_EVENTS_ROUTE,
+  ACTIONS_PAGE_ROUTE,
+  COIN_MACHINE_ROUTE,
+  USER_ROUTE,
+  MEMBERS_ROUTE,
+  COLONY_EXTENSION_SETUP_ROUTE,
+  COLONY_EXTENSION_DETAILS_ROUTE,
+  COLONY_EXTENSIONS_ROUTE,
+  COLONY_FUNDING_ROUTE,
+} from '~routes/routeConstants';
+import { SimpleMessageValues } from '~types/index';
+import { useColonyFromNameQuery } from '~data/index';
+
+const MSG = defineMessages({
+  connect: {
+    id: 'routeTitles.connect',
```

Comment:
I am not sure that these should be the ids. I would think it should be `utils.hooks.useTitle.connect` as we give it based on the folder location and file name. Here is an example from the `BuyTokens` component:
` id: 'dashboard.CoinMachine.BuyTokens.title'`

I would double check with Raul...

Reply:
I agree with you. I was a little confused since there was no other example of messages being inside utils hooks. I will change them quick.